### PR TITLE
[WIP] Improve custom fact handling

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: moduleroot/spec/facts/puppetlabs/stdlib.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+add_custom_fact :puppet_environmentpath, '/etc/puppetlabs/code/environments'
+add_custom_fact :puppet_vardir, '/opt/puppetlabs/puppet/cache'
+add_custom_fact :root_home, '/root'
+add_custom_fact :service_provider, ->(os, facts) do
```
Comment (bastelfreak) [inline moduleroot/spec/facts/puppetlabs/stdlib.rb]: what does `->` do and how is this operator called?

**Comment by bastelfreak** [comment]
Thanks for the PR. We should test it on a module to see that it actually works (and call it acceptance-testing for msync)

File: moduleroot/spec/facts/puppetlabs/stdlib.rb (hunk 1 above)
Comment (ekohl) [reply moduleroot/spec/facts/puppetlabs/stdlib.rb]: I based it on https://github.com/mcanevet/rspec-puppet-facts/blob/fe21de12108fbab0123bdc4db2365a29ea62f171/spec/rspec_puppet_facts_spec.rb#L576-L580 so I guess it's called a proc :)

File: moduleroot/spec/facts/puppetlabs/stdlib.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  case facts[:osfamily]
+  when 'RedHat'
+    facts[:operatingsystemrelease].to_i >= 7 ? 'systemd' : 'sysv'
```
Comment (ekohl) [inline moduleroot/spec/facts/puppetlabs/stdlib.rb]: I was also offline so I don't know if it's actually sysv or something else.

**Comment by ekohl** [comment]
What I used to verify it:

```
require 'spec_helper'

on_supported_os.each do |os, facts|
  context "on #{os}" do
    it { expect(facts["service_provider"]).not_to be_nil }
  end
end
```

File: moduleroot/spec/facts/puppetlabs/stdlib.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+add_custom_fact :puppet_environmentpath, '/etc/puppetlabs/code/environments'
+add_custom_fact :puppet_vardir, '/opt/puppetlabs/puppet/cache'
+add_custom_fact :root_home, '/root'
```
Comment (juniorsysadmin) [inline moduleroot/spec/facts/puppetlabs/stdlib.rb]: Do we need to account for FreeBSD/OpenBSD tests?

Comment (ekohl) [reply moduleroot/spec/facts/puppetlabs/stdlib.rb]: Maybe. I just went through what some modules set as standards but they probably disregarded those as well. Also no idea what it does on Windows.

**Comment by ghoneycutt** [comment]
Seems we already have this functionality with https://github.com/mcanevet/rspec-puppet-facts 

https://github.com/voxpupuli/modulesync_config/blob/master/config_defaults.yml#L39

**Comment by ekohl** [comment]
No, we don't. Those are the built in Puppet facts. This is intended to mock custom facts from modules. I'm still not sure if I like this format. Ideally the mocked fact could live inside the modules themselves so we don't need to modulesync it everywhere.

**Comment by ghoneycutt** [comment]
Closing due to inactivity, please feel free to re-open.

**Pull request closed without merging**